### DISCUSSION TITLE
Is there a way to turn off/reduce autocomplete?

### DISCUSSION THREAD

**Discussion by Satvik** [fonsp/Pluto.jl#3001] [Questions about using Pluto] [answered]
Pluto seems to fill in autocomplete suggestions on a lot of triggers, which doesn't quite work with my workflow. For example, say I have

```
bits = [0, 0b1, 0b10]
bits_file = "my_filename.txt"
```

and I type `f(bits` . Pluto's autocomplete suggests `bits_file` and will fill it in when I type the closing parenthesis. 

Is there a way to change this? Ideally I would want autocomplete suggestions to only be accepted if I type tab or enter.

**Comment by greimel** [comment]
I realized that as well. In my opinion Pluto shouldn't actually execute the completion on typing, but require an explicit `<TAB>` or similar.

**Reply by fonsp** [reply, marked as answer]
This behaviour was changed in https://github.com/fonsp/Pluto.jl/pull/3055

**Comment by disberd** [comment]
@Satvik, @greimel Have a look at #3008 as you might realize the cause and at least the temporary solution

**Reply by Satvik** [reply]
This worked for me, thanks!

**Comment by fonsp** [comment]
Hey @Satvik !

Thanks for your feedback! I can't reproduce your example, but I'm curious to hear more examples where it's not working as expected!

<img width="408" alt="image" src="https://github.com/user-attachments/assets/7fb8b892-ca85-4c37-946d-13d03dface8c">

**Comment by greimel** [comment]
You can't write `f(x,y,z; kwargs...) = 2x`. Instead

https://github.com/user-attachments/assets/3cdf4eb0-eb8b-47dc-8431-687f680b20bb

**Reply by disberd** [reply]
Basically the same example I put here #3011 (but yours is even better at conveying the message :D)

**Reply by greimel** [reply]
sorry, hadn't seen that! didn't intend to plagiarize ;-)

**Reply by disberd** [reply]
Ahah i did not write the comment to imply that, just to point to the associated issue (and note that f(x, y) is an even better example)

**Reply by fonsp** [reply]
This has been fixed in https://github.com/fonsp/Pluto.jl/pull/3055

**Comment by greimel** [comment]
Another example is working with symbols is also quite annoying

https://github.com/user-attachments/assets/7ddfe0b2-480a-4df9-b273-83ac1ad728ee

**Reply by fonsp** [reply]
Fixed in https://github.com/fonsp/Pluto.jl/commit/090a9a9c2ce1a1c3c6cfcb516d0af0faa0b98249

**Comment by greimel** [comment]
also, it seems that the caching of symbols takes some time. Sometimes `data(` expands to `DataFrame(` at the first try, but doesn't anymore later on.

**Reply by fonsp** [reply]
This has been fixed in https://github.com/fonsp/Pluto.jl/pull/3055
